### PR TITLE
Implemented custom theme support

### PR DESCRIPTION
**Pull request by Qrivi**
Rather than relying on only Material widgets that can switch between a light an a dark mode, and having to tap into very restricted properties of `ThemeData` using `Theme.of(context)`, this PR adds a `CustomTheme` class on which we can set all colors we use throughout the app's design. This also allows for multiple themes -- we are not restricted to only 1 light mode and 1 dark mode.

`CustomTheme` has a getter `themeData` which generates `ThemeData` that can be passed to the `MaterialApp` to theme Material widgets. Custom widgets however should no longer use `Theme.of(context)` but `ref.watch(settingsControllerProvider).activeTheme`, which returns a `CustomTheme`. Aside from keeping a reference to the `activeTheme`, also keeps references of the user's preferred light and dark themes (in case in the future we implement multiple themes and allow the user to choose -- see popular apps like Twitter or Reddit that have eg a regular dark mode and an amoled dark mode).

Finally, this controller also fixes the status bar on iOS (and should do too on Android, but cannot test as this needs to be checked on a physical device) when resolving the theme, and has methods `invertStatusBarColors()` and `resetStatusBarColors()` in case we want to temporarily set the opposite status bar colors on some views.

Tested to show correct status bar colors on iOS 15.2 when
- launching the app
- toggling dark mode in the settings app with Rescado suspended
- toggling dark mode using Control Center
- toggling dark mode using Siri

This should fix issue #3 for good. 🙂

https://user-images.githubusercontent.com/3763638/148833662-a392fcfc-18c1-4389-a21f-ba345df8e110.mp4

### CODE REVIEW

File: lib/src/constants/rescado_constants.dart [hunk 1]
```diff
@@ -1,3 +1,6 @@
+import 'package:flutter/cupertino.dart';
```
Comment (Qrivi) [inline lib/src/constants/rescado_constants.dart]: No cupertino. 😩

File: lib/src/views/splash_view.dart [hunk 2]
```diff
@@ -46,65 +47,58 @@ class _SplashViewState extends State<SplashView> with SingleTickerProviderStateM
...
-                      data: (User user) {
-                        var nextViewId = (user.status == UserStatus.anonymous || user.status == UserStatus.identified) ? MainView.viewId : AuthenticationView.viewId;
-                        // Future.delayed(Duration.zero) is a bit too hacky to my liking but we need to return a widget and using Navigator otherwise shortcuts the flow, resulting in errors.
-                        Future.delayed(Duration.zero, () => Navigator.pushReplacementNamed(context, nextViewId));
-                        return _buildPlaceholder();
-                      },
-                      error: (Object error, _) {
-                        _animationController.reset();
-                        final isOffline = error is OfflineException;
+            Consumer(
```
Comment (Pekket) [inline lib/src/views/splash_view.dart]: `Consumer` can be removed since the view became a `ConsumerState`.
You can use the global `ref.watch` 🧞

Comment (Qrivi) [reply lib/src/views/splash_view.dart]: Fixed in latest commit. :)

**Review by Pekket — APPROVED** [review]
🧞

**Pull request merged**